### PR TITLE
composer test is now more convenient to use

### PR DESCRIPTION
**Pull request by ipundit**
composer test now takes optional parameters

```php
composer test [fileName [filter]]
```
### With no parameters
```php
composer test
```
runs
```php
vendor/bin/phpunit
```
### With a file name
```php
composer test dateTime
```
runs
```php
vendor/bin/phpunit test/DateTimeTest.php
```
Note the prepending of test/, the fix in capitalization, and appending Test.php
### With a file name and filter
```php
composer test dateTime format
```
runs
```php
vendor/bin/phpunit --filter format test/DateTimeTest.php
```

### CODE REVIEW

**Comment by shmax** [comment]
Aren't you kind of reinventing the wheel, here? Isn't a lot of this already possible using the `--` pass-through? eg:

```sh
composer test -- test/DateTimeTest.php
```

if you don't like typing the name of the folder or ".php", you can do this:
```sh
composer test -- --filter DateTimeTest
```

single test:
```
composer test -- --filter DateTimeTest::testSetIsoDate
```

**Comment by ipundit** [comment]
You can still do that if you want after this commit: 

```php
vendor/bin/phpunit --filter DateTimeTest::testSetIsoDate
```

but, this is more convenient:

```php
composer test dateTime setIsoDate
```

It's a few things less to type and it's easier to do progressively larger tests. In my dev cycle, I do:

```php
composer test dateTime setIsoDate
composer test dateTime
composer test
```
In the old way, you'd have to:
```php
composer test -- --filter DateTimeTest::testSetIsoDate
composer test -- --filter DateTimeTest
composer test
```
Which would you rather type? Which is easier for a new developer to learn, especially since running:
```php
composer test
```
gives hints for:
```php
To run just the tests in test/CallbackTest.php, try: composer test callback
To run a specific test in test/DateTimeTest.php, try: composer test dateTime testSetIsoDate
```
Discoverability is important for new developers, and convenience, especially if it's optional to use, is useful for all developers.

**Comment by shmax** [comment]
I dunno, I'm still not convinced. Developers already accustomed to using the standard phpunit API now have to learn yours, so I guess it's up for debate which is more convenient.

And for what it's worth, I never use the CLI for tests; I use PHPStorm, and I can just right click on folders, test classes, or individual tests and choose "Run" or "Debug".

If you really want to insist on it I guess we can merge this, but I think it would be more appropriate for you to develop this tool in its own composer package, and we can import it as a dependency once it's matured a bit.

**Comment by ipundit** [comment]
All these tests now work too:
```
composer test test/DateTimeTest.php
composer test -- test/DateTimeTest.php
composer test -- --filter DateTimeTest
composer test -- --filter DateTimeTest::testSetIsoDate
```

As well as:
```
composer test dateTime setIsoDate
composer test dateTime
composer test
```

Merge and let future developers use whatever syntax they want to use? Developers can write personal Windows.bat files or shell files if they wish, but using `composer test` makes the convenience scripts more cross-platform.

**Comment by shmax** [comment]
I think I can live with that. Please fix the CI, and we'll move forward.

File: composer.json [hunk 1]
```diff
@@ -22,17 +22,31 @@
       "lib/Utils.php"
     ],
     "psr-4": {
-      "ActiveRecord\\": "lib/",
+      "ActiveRecord\\": "lib/"
+    }
+  },
+  "autoload-dev": {
```
Comment (shmax) [inline composer.json:28]: Oh, nice, I was wondering what to do about that 👍

**Review by shmax — APPROVED** [review]
Nice 👍

**Pull request merged**